utils/funcexp/func_time_to_sec.cpp

```diff
@@ -194,4 +210,44 @@ int64_t Func_time_to_sec::getIntVal(rowgroup::Row& row, FunctionParm& parm, bool
   return rtn;
 }
 
+IDB_Decimal Func_time_to_sec::getDecimalVal(rowgroup::Row& row, FunctionParm& parm, bool& isNull,
+                                            CalpontSystemCatalog::ColType& op_ct)
+{
+  int64_t value = getIntVal(row, parm, isNull, op_ct);
+  IDB_Decimal decimal;
+  int32_t scaleDiff = 6 - op_ct.scale;
+
+  if (!op_ct.isWideDecimalType())
+  {
+    if (scaleDiff > 0)
+    {
+      value = (int64_t)(value > 0 ? (double)value / IDB_pow[scaleDiff] + 0.5
+                                  : (double)value / IDB_pow[scaleDiff] - 0.5);
+    }
+    else if (scaleDiff < 0)
+    {
+      value = value * IDB_pow[-scaleDiff];
+    }
+    decimal.value = value;
+  }
+  else
+  {
+    int128_t s128Value = value;
+    if (scaleDiff > 0)
+    {
+      s128Value = (int128_t)(s128Value > 0 ? (double)s128Value / IDB_pow[scaleDiff] + 0.5
```

Comment:
Here can be a loss of precision. Better cast to long double. It is as good as double in worst case but certainly can be beneficial when it is present.

Also, there is a nice function round: https://cplusplus.com/reference/cmath/round/

It does more stable rounding than what is in code (a/b + 0.5).

Reply:
This issue has been fixed.